File: conformance/bundles/06-doc-string-mismatch/echo.steps.rs
//! Rust sibling of `echo.steps.ts` (bundle `06-doc-string-mismatch`).

use varar::Steps;

pub fn register(s: &mut Steps<()>) {
    // Returns the WRONG string for the doc-string slot; the core compares it
    // and fails with DocStringMismatch.
    s.sensor("I echo the following:", |_ctx: (), _doc: String| Ok("goodbye".to_string()));
}

pub fn state() {}

File: conformance/bundles/06-doc-string-mismatch/echo.steps.go
// Go sibling of echo.steps.ts (bundle 06-doc-string-mismatch).
package fixture

import "github.com/varar-dev/varar-go/varar"

func Register(s *varar.Steps) {
	// Returns the WRONG string (bare — the doc string is the only slot); the
	// core compares it to the doc string and fails with DocStringMismatch.
	s.Sensor("I echo the following:", func(state varar.Value, args []varar.Value) (*varar.Value, error) {
		return varar.Ptr(varar.StrValue("goodbye")), nil
	})
}

func State() varar.Value {
	return varar.NullValue
}
